### PR TITLE
[Feature][Conenctor-V2] Add openmldb source connector

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

Add openmldb source connector #3018 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
openmldb only support these data types:

image

spark integration test:

image

flink integration test:

image

image

**Review by ashulin — COMMENTED** [review]
Can you add an e2e use case？

**Comment by TyrantLucifer** [comment]
> Can you add an e2e use case？

OpenMldb image need change ip address that in config files. Ip can not be code name，so that use test container to write e2e is impossible and I desided to do the integration test instead of it.

**Comment by TaoZex** [comment]
> > Can you add an e2e use case？
> 
> OpenMldb image need change ip address that in config files. Ip can not be code name，so that use test container to write e2e is impossible and I desided to do the integration test instead of it.

Doris was the same way. I was trying to automatically change ip address through shell script when docker was opened.

**Comment by TyrantLucifer** [comment]
> > > Can you add an e2e use case？
> > 
> > 
> > OpenMldb image need change ip address that in config files. Ip can not be code name，so that use test container to write e2e is impossible and I desided to do the integration test instead of it.
> 
> Doris was the same way. I was trying to automatically change ip address through shell script when docker was opened.

Test container uses hostname to assign ip to container, how do I get the real ip address and change it in container? If you have good ideas to solve this problem, can you do this work or give some guides to me?

**Comment by TaoZex** [comment]
> > > > Can you add an e2e use case？
> > > 
> > > 
> > > OpenMldb image need change ip address that in config files. Ip can not be code name，so that use test container to write e2e is impossible and I desided to do the integration test instead of it.
> > 
> > 
> > Doris was the same way. I was trying to automatically change ip address through shell script when docker was opened.
> 
> Test container uses hostname to assign ip to container, how do I get the real ip address and change it in container? If you have good ideas to solve this problem, can you do this work or give some guides to me?

I'm trying to do this, and I'll let you know if I get any good news.

File: docs/en/connector-v2/source/OpenMldb.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+> OpenMldb source connector
+
+## Description
+
+Used to read data from OpenMldb.
+
+## Key features
+
+- [x] [batch](../../concept/connector-v2-features.md)
+- [x] [stream](../../concept/connector-v2-features.md)
```
Comment (hailin0) [inline docs/en/connector-v2/source/OpenMldb.md:12]: Please check support stream read mode？

File: seatunnel-connectors-v2/connector-openmldb/src/main/java/org/apache/seatunnel/connectors/seatunnel/openmldb/source/OpenMldbSource.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            this.seaTunnelRowType = convert(columnList);
+        } catch (SQLException | SqlException e) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, "Failed to initialize data schema");
+        }
+    }
+
+    @Override
+    public Boundedness getBoundedness() {
+        return JobMode.BATCH.equals(jobContext.getJobMode()) ? Boundedness.BOUNDED : Boundedness.UNBOUNDED;
+    }
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-openmldb/src/main/java/org/apache/seatunnel/connectors/seatunnel/openmldb/source/OpenMldbSource.java:107]: Please check support stream read mode？

**Review by hailin0 — COMMENTED** [review]
Please implement `TableSourceFactory`

**Review by EricJoy2048 — APPROVED** [review]

**Review by TaoZex — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]
LGTM, It looks good overall

**Pull request merged**